aucpace/examples/key_agreement.rs

```diff
@@ -0,0 +1,315 @@
+use aucpace::{Client, ClientMessage, Database, Result, Server, ServerMessage};
+use curve25519_dalek::ristretto::RistrettoPoint;
+use password_hash::{ParamsString, SaltString};
+use rand_core::OsRng;
+use scrypt::{Params, Scrypt};
+use sha2::digest::Output;
+use sha2::Sha512;
+use std::io::{Read, Write};
+use std::net::{IpAddr, Ipv4Addr, SocketAddr, TcpListener, TcpStream};
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::thread;
+use std::time::Instant;
```

Comment:
Most of our other crates support `no_std`, in most cases with heapless functionality.

It's fine if `alloc` is a hard dependency, but would it be possible to make the `std`-dependent features optional?